zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BasicAuthInterceptor.java

```diff
@@ -65,24 +64,23 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         if (!msg.status().equals(HttpStatus.FORBIDDEN) || content.isEmpty()) {
           return HttpResponse.of(msg);
         }
-        try {
-          final ByteBuffer buf;
-          if (content instanceof ByteBufHolder) {
-            buf = ((ByteBufHolder) content).content().nioBuffer();
-          } else {
-            buf = ByteBuffer.wrap(content.array());
-          }
+        final ByteBuf buf;
+        if (content instanceof ByteBufHolder) {
+          buf = ((ByteBufHolder) content).content();
+        } else {
+          buf = Unpooled.wrappedBuffer(content.array());
+        }
+        try (ByteBufInputStream stream = new ByteBufInputStream(buf, true)) {
           try {
-            JsonReader message = enterPath(JsonReader.of(
-              Okio.buffer(Okio.source(ReadBuffer.wrapUnsafe(buf)))), "message");
-            if (message != null) throw new IllegalStateException(message.nextString());
+            JsonParser message = enterPath(JsonSerializers.jsonParser(stream), "message");
+            if (message != null) throw new IllegalStateException(message.getValueAsString());
           } catch (IOException e) {
             Exceptions.throwUnsafely(e);
             throw new UncheckedIOException(e);  // unreachable
           }
           throw new IllegalStateException(msg.toString());
-        } finally {
-          ReferenceCountUtil.safeRelease(content);
```

Comment:
no need to release anymore?

Reply:
Yeah, we pass `releaseOnClose = true` to `ByteBufInputStream`'s constructor instead.